physx/src/controller.rs

```diff
@@ -0,0 +1,103 @@
+#![warn(clippy::all)]
+#![warn(rust_2018_idioms)]
+
+use crate::px_type::*;
+use crate::rigid_dynamic::RigidDynamic;
+
+use glam::Vec3;
+use physx_macros::*;
+
+use physx_sys::{
+    phys_PxCreateControllerManager, PxCapsuleControllerDesc, PxCapsuleControllerDesc_delete,
+    PxCapsuleControllerDesc_isValid, PxCapsuleControllerDesc_new_alloc, PxController,
+    PxControllerDesc, PxControllerManager, PxControllerManager_createController_mut,
+    PxController_getActor, PxController_getPosition, PxController_release_mut,
+    PxController_setPosition_mut, PxExtendedVec3, PxMaterial, PxScene,
+};
+
+#[physx_type]
+impl ControllerDesc {}
+
+#[physx_type]
+impl ControllerManager {
+    pub fn new(scene: *mut PxScene) -> Self {
+        let ptr = unsafe { phys_PxCreateControllerManager(scene, false) };
+        ControllerManager { ptr }
+    }
+
+    pub fn create_controller(&self, desc: &mut ControllerDesc) -> Controller {
+        let controller =
+            unsafe { PxControllerManager_createController_mut(self.ptr, desc.get_raw_mut()) };
```

Comment:
The desc here should be just `get_raw()` here, it's a `*const PxControllerDesc`.